internal/secrets/localsecrets/localsecrets.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localsecrets
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"golang.org/x/crypto/nacl/secretbox"
+	"io"
+)
+
+// secretKeeper holds a secret for use in symmetric encryption,
+// and implementations of driver.Encryper and driver.Decrypter.
+type secretKeeper struct {
+	secretKey [32]byte
+	Encrypter driver.Encrypter
+	Decrypter driver.Decrypter
+}
+
+// NewSecretKeeper takes a secret key and returns a secretKeeper.
+func NewSecretKeeper(sk string) (*secretKeeper, error) {
+	skb := []byte(sk)
+	enc := base64.StdEncoding
+	dst := make([]byte, enc.EncodedLen(len(skb)))
+	enc.Encode(dst, skb)
+
+	var dst32 [32]byte
+	copy(dst32[:], dst)
+
+	skr := &secretKeeper{secretKey: dst32}
+	skr.Encrypter = &encrypter{skr: skr}
+	skr.Decrypter = &decrypter{skr: skr}
+
+	return skr, nil
+}
+
+// encrypter implemets driver.Encrypter and holds a pointer to
+// a secretKeeper, which holds the secret.
+type encrypter struct {
+	skr *secretKeeper
+}
+
+// Encrypt encrypts a message using a per-message generated nonce and
+// the secret held in the encrypter's secretKeeper.
+func (e *encrypter) Encrypt(ctx context.Context, message []byte) ([]byte, error) {
+	var nonce [24]byte
+	if _, err := io.ReadFull(rand.Reader, nonce[:]); err != nil {
+		return nil, err
+	}
+	// a slice beginning at nonce is used here as the destination for the encrypted message,
```

Comment:
Punctuate this as a sentence.

Also, it confuses me. The 24-byte `nonce` array could be too small for the encrypted message. And how does making nonce the destination have anything to do with the first 24 bytes of the encrypted message?

Obviously I'm very confused about how `secretbox.Seal` works, and unfortunately your comment doesn't help me.

Reply:
Ah, I read the Seal docs and realized what's going on here. Your code is fine and this is mostly my problem. 

Maybe you could add something that says that Seal appends the encrypted message to its first argument and returns the result.